libjob_queue/src/lsf_driver.c

```diff
@@ -228,6 +228,51 @@ int lsf_job_parse_bsub_stdout(const char * bsub_cmd, const char * stdout_file) {
   return jobid;
 }
 
+/**
+ * Assumes fname points to a file with content "hname1:hname2:hname3" as written by lsf_job_write_bjobs_to_file
+ */
+stringlist_type * lsf_job_parse_bjobs_file(const char* fname) {
```

Comment:
1. This function can either return `NULL` or a `stringlist_type` instance; safer to return an empty list than `NULL`.
2. Functions which allocate memory which must be later free'd should have `alloc` as part of their name.
3. The purpose of this function is to extract hostnames, not really general bjobs parsing - I would suggest e.g. `lsf_job_parse_hostnames`.